keyvault/client/src/main/java/com/azure/keyvault/models/SecretInfo.java

```diff
@@ -0,0 +1,347 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Map;
+
+/**
+ * SecretInfo is the resource containing all the properties of the secret except its value.
+ * It is managed by the Secret Service.
+ */
+public class SecretInfo {
+
+    /**
+     * The secret id.
+     */
+    private String id;
+
+    /**
+     * The secret version.
+     */
+    private String version;
+
+    /**
+     * Determines whether the object is enabled.
+     */
+    private Boolean enabled;
+
+    /**
+     * Not before date in UTC.
+     */
+    private OffsetDateTime notBefore;
+
+    /**
+     * Expiry date in UTC.
+     */
+    private OffsetDateTime expires;
+
+    /**
+     * Creation time in UTC.
+     */
+    private OffsetDateTime created;
+
+    /**
+     * Last updated time in UTC.
+     */
+    private OffsetDateTime updated;
+
+    /**
+     * The secret name.
+     */
+    private String name;
+
+    /**
+     * Reflects the deletion recovery level currently in effect for secrets in
+     * the current vault. If it contains 'Purgeable', the secret can be
+     * permanently deleted by a privileged user; otherwise, only the system can
+     * purge the secret, at the end of the retention interval. Possible values
+     * include: 'Purgeable', 'Recoverable+Purgeable', 'Recoverable',
+     * 'Recoverable+ProtectedSubscription'.
+     */
+    private String recoveryLevel;
+
+
+    /**
+     * The content type of the secret.
+     */
+    @JsonProperty(value = "contentType")
+    private String contentType;
+
+
+    /**
+     * Application specific metadata in the form of key-value pairs.
+     */
+    @JsonProperty(value = "tags")
+    private Map<String, String> tags;
+
+    /**
+     * If this is a secret backing a KV certificate, then this field specifies
+     * the corresponding key backing the KV certificate.
+     */
+    @JsonProperty(value = "keyId", access = JsonProperty.Access.WRITE_ONLY)
```

Comment:
the json key for keyId is actually "kid" when sent across the wire.

Reply:
True. updated.